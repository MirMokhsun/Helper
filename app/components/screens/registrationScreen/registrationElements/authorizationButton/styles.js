import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    wrapper: {
        flexDirection: 'row',
        height: 30,
        paddingTop: 10,
    },
    link: {
        marginLeft: 5,
        color: '#1459D2',
        fontSize: 18,
        fontFamily: 'Roboto',
        fontWeight: 'bold',
        lineHeight: 18
    },
    text: {
        color: '#000756',
        fontSize: 18,
        fontFamily: 'Roboto',
        lineHeight: 18
    }
});