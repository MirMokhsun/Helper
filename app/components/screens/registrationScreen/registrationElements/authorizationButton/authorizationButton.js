import React, { Component } from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { styles } from './styles';
import { withNavigation } from 'react-navigation';

export class AuthorizationButton extends Component {
    constructor(props){
        super(props)
    }

    routeAuthorization = () => {
        const { navigation } = this.props;
        navigation.navigate('AuthorizationScreen')
    }

    render() {
        const { text, textLogin } = this.props;
        return (
            <View style={styles.wrapper}>
                <Text style={styles.text}>{text}</Text>
                <TouchableOpacity onPress={this.routeAuthorization} testID="BtnAuthorizationInRegistrationID" accessibilityLabel="BtnAuthorizationInRegistrationID">
                    <View >
                        <Text style={styles.link}>{textLogin}</Text>
                    </View>
                </TouchableOpacity >
            </View>
        )
    }
}

export default withNavigation(AuthorizationButton);