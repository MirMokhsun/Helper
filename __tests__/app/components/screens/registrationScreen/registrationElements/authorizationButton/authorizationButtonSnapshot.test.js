import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { AuthorizationButton } from '../../../../../../../app/components/screens/registrationScreen/registrationElements/authorizationButton/authorizationButton';

it('create snapshot AuthorizationButton', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<AuthorizationButton
        text={"Already have an account?"}
        textLogin={'login'}
        navigation={() => { }}
    />);
    expect(toolbarSnapshot).toMatchSnapshot();
});