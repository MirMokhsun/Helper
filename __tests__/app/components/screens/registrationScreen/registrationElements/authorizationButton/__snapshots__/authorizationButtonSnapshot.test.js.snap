// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot AuthorizationButton 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 30,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000756",
        "fontFamily": "Roboto",
        "fontSize": 18,
        "lineHeight": 18,
      }
    }
  >
    Already have an account?
  </Text>
  <TouchableOpacity
    accessibilityLabel="BtnAuthorizationInRegistrationID"
    activeOpacity={0.2}
    onPress={[Function]}
    testID="BtnAuthorizationInRegistrationID"
  >
    <View>
      <Text
        style={
          Object {
            "color": "#1459D2",
            "fontFamily": "Roboto",
            "fontSize": 18,
            "fontWeight": "bold",
            "lineHeight": 18,
            "marginLeft": 5,
          }
        }
      >
        login
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;
